core/client-internal/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -72,6 +72,9 @@ public AlluxioBlockStore getAluxioBlockStore() {
    * {@link ClientContext}.
    */
   public void reset() {
+    if (null != mFileSystemMasterClientPool) {
```

Comment:
Can you explain why this is needed? When the context resets, the pool might need reconstruction, too, since the master address can change.